modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/GridCacheSetItemKey.java

```diff
@@ -85,7 +85,10 @@ public GridCacheSetItemKey() {
 
         GridCacheSetItemKey that = (GridCacheSetItemKey)o;
 
-        return setId.equals(that.setId) && item.equals(that.item);
+        if ((setId == null && that.setId != null) || (setId != null && that.setId == null))
```

Comment:
may be         Objects.equals(setId, that.setId); ?

Reply:
Simplified